lib/cyanide.ex

```diff
@@ -157,9 +157,14 @@ defmodule Cyanide do
   # TODO: Cyanide 2.0: incompatible change, use %Cyanide.Binary{} instead
   defp parse_value(0x5, map, key, <<subdoc_size::little-32, subtype::8, subdoc_and_rest::binary>>) do
     with the_size when the_size >= 0 <- subdoc_size,
-         <<subdocument::binary-size(the_size), rest::binary>> <- subdoc_and_rest do
-      Map.put(map, key, {subtype, subdocument})
-      |> parse_doc_bytes(rest)
+         <<subdocument::binary-size(the_size), rest::binary>> <- subdoc_and_rest,
+         subtype_atom <- Binary.subtype_int_to_atom(subtype) do
```

Comment:
Instead of nesting a call that can fail, split it using the `with` construct:
```
with the_size when the_size >= 0 <- subdoc_size,
             <<subdocument::binary-size(the_size), rest::binary>> <- subdoc_and_rest,
             {:ok, subtype} <- Binary.subtype_int_to_atom(subtype) do
           [...]
```
Have a look [here](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#with/1) for a complete reference